core/schemas/observables/user_account.py

```diff
@@ -0,0 +1,32 @@
+import datetime
+from typing import Literal
+
+from core.schemas import observable
+from pydantic import model_validator
+
+
+class UserAccount(observable.Observable):
+    type: Literal[observable.ObservableType.user_account] = observable.ObservableType.user_account
+    user_id: str | None = None
+    credential: str | None = None
+    account_login: str | None = None
+    account_type: str | None = None
+    display_name: str | None = None
+    is_service_account: bool | None = None
+    is_privileged: bool | None = None
+    can_escalate_privs: bool | None = None
+    is_disabled: bool | None = None
+    account_created: datetime.datetime | None = None
+    account_expires: datetime.datetime | None = None
+    credential_last_changed: datetime.datetime | None = None
+    account_first_login: datetime.datetime | None = None
+    account_last_login: datetime.datetime | None = None
+
+    @model_validator(mode='after')
+    def check_timestamp_coherence(self) -> 'UserAccount':
+        if self.account_created and self.account_expires:
+            if self.account_created > self.account_expires:
+                raise ValueError("Account created date is after account expiration date.")
+        return self
```

Comment:
Could we add:

* A docstring with an explanation of all these fields, or at least a reference to a canoncal source that explains how to use them
* Validating that at least some fields are set? It's currently possible to create a user_account where everything is None.
* Also, what would `value` be set to here when creating one of these? This should also be explained in the docstring